### DISCUSSION TITLE
Broken URL

### DISCUSSION THREAD

**Discussion by jpines-vi** [aws/amazon-neptune-jdbc-driver#219] [General]
I started writing a bug for the SQL-Gremlin implementation when I saw a reference to [this URL](https://github.com/aws/amazon-neptune-jdbc-driver/blob/develop/sql-gremlin/README.asciidoc#sql-support-and-limitations). Unfortunately, that URL is broken so I cannot tell if my issue is known or not. Can someone correct that URL for me so I can be educated before writing the bug?

**Comment by xiazcy** [comment]
Hello, thank you for reporting this. 

Yes, it was removed unintentionally. We have a PR to add it back, https://github.com/aws/amazon-neptune-jdbc-driver/pull/217. We need to get the license updated in order to get the PR merged, which should be done sometimes this week. 

If you'd like to see the limitations for now, you can checkout the file here: https://github.com/aws/amazon-neptune-jdbc-driver/blob/8547aabdf822b58017a57245d4346c3267775867/markdown/sql-gremlin.md.

**Reply by jpines-vi** [reply]
Thank you! 

I wonder though it is not explicitly stated, would you consider the fact that the SQL-Gremlin driver has to scan the whole graph to build a schema a limitation? I ask because it turns out that my graph of 3,772,891 vertices and 7,615,616 edges times out while trying to fetch the metadata. I can avoid that by supplying my own schema for a pure Gremlin query but the SQL-Gremlin query does not seem to allow for that, always fetching the schema when running a query.

**Reply by jpines-vi** [reply]
To be clear, I do see that Schema Collection is mentioned as a limitation but it is unclear to me what that limitation entails, just that it does it and how it formats the results.

**Reply by xiazcy** [reply]
That is correct, currently there is no way for users to supply their own schema. I do agree that using a supplied schema would improve performance. I believe the original intent is to keep the driver simple for users, so we are looking in the direction of improving schema collection instead, but feel free to open a feature request for supplying schema. 

Thank you!
